lib/compressMeshes.js

```diff
@@ -0,0 +1,518 @@
+        max : indicesAccessor.max,
+        min : indicesAccessor.min,
+        type : indicesAccessor.type
+    };
+    var indicesAccessorId = addToArray(gltf.accessors, newIndicesAccessor);
+    primitive.indices = indicesAccessorId;
+
+    // Remove attributes bufferViews.
+    for (var semantic in primitive.attributes) {
+        if (primitive.attributes.hasOwnProperty(semantic)) {
+            var attributeAccessor = gltf.accessors[primitive.attributes[semantic]];
+            var newAttributeAccessor = {
+                componentType : attributeAccessor.componentType,
+                count : attributeAccessor.count,
+                max : attributeAccessor.max,
+                min : attributeAccessor.min,
+                type : attributeAccessor.type
+            };
+            var attributeAccessorId = addToArray(gltf.accessors, newAttributeAccessor);
+            primitive.attributes[semantic] = attributeAccessorId;
+        }
+    }
+
+    var buffer = {
+        byteLength : encodedLen,
+        extras : {
+            _pipeline : {
+                extension : '.bin',
+                source :encodedData
+            }
+        }
+    };
+    var bufferId = addToArray(gltf.buffers, buffer);
+    var bufferView = {
+        buffer : bufferId,
+        byteOffset : 0,
+        byteLength : encodedLen
+    };
+    var bufferViewId = addToArray(gltf.bufferViews, bufferView);
+
+    var extensions = primitive.extensions;
+    if (!defined(primitive.extensions)) {
+        extensions = {};
+        primitive.extensions = extensions;
+    }
+    var draco_extension = {
+        bufferView : bufferViewId,
+        attributes : attributeToId,
+    };
+    extensions.KHR_draco_mesh_compression = draco_extension;
+}
+
+function copyCompressedExtensionToPrimitive(primitive, compressedPrimitive) {
+    var attributes = {};
+    for (var semantic in primitive.attributes) {
+        if (primitive.attributes.hasOwnProperty(semantic)) {
+            attributes[semantic] = compressedPrimitive.attributes[semantic];
+        }
+    }
+    primitive.attributes = attributes;
+    primitive.indices = compressedPrimitive.indices;
+
+    var draco_extension = compressedPrimitive.extensions.KHR_draco_mesh_compression;
+    var extensions = {};
+    primitive.extensions = extensions;
+    var copied_extension = {
+        bufferView : draco_extension.bufferView,
+        attributes : draco_extension.attributes,
+    };
+    extensions.KHR_draco_mesh_compression = copied_extension;
+}
+
+/**
+ * Compresses all the meshes using Draco compression.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset.
+ * @param {Object} options Main options object.
+ * @returns {Object} The glTF asset with compressed meshes.
+ *
+ * @private
+ */
+function compressMeshes(gltf, options) {
+    addExtensionsRequired(gltf, 'KHR_draco_mesh_compression');
+    var hashPrimitives = [];
+    options = options === undefined ? {} : options;
```

Comment:
For better usage, I think this should be:
```
options = defaultValue(options, {});
options.dracoOptions = defaultValue(options.dracoOptions, {});
```

Otherwise the `options.dracoOptions.property` will result in `dracoOptions` being undefined.

Reply:
Done.